### ISSUE TITLE
[FEATURE] Implement Dark Mode Theme Support.

### ISSUE THREAD

**Issue by prashantrai-30** [asyncapi/modelina#2120]
### Why do we need this improvement?

- **Current Limitations** ⚠️
  - Only light mode available
  - Difficult to use in low-light environments
  - Limited accessibility options

- **User Feedback** 👥
  - Frequent requests for dark mode
  - Reports of eye strain during night usage
  - Accessibility concerns from users

- **Market Analysis** 📊
  - Standard feature in competitor APIs
  - Present in major development tools
  - Growing user expectation

### How will this change help?

- **Solution Impact** 🎯
  - System-level theme detection
  - Seamless mode switching
  - Persistent user preferences

- **Direct Benefits** ⚡
  - Reduced eye strain in low light
  - Better readability at night
  - Lower battery consumption on OLED displays

- **Implementation Value** 🔨
  - Standardized color variables
  - Consistent UI components
  - Future-proof theming system
  
- **Business Metrics** 📈
  - Increased user satisfaction
  - Better platform accessibility
  - Improved market competitiveness

### How could it be implemented/designed?

- Create theme context/provider
- Define CSS variables for theming
- Add theme toggle component
- Implement system theme detection
- Add local storage persistence
- Update existing components to use theme variables

### 🚧 Breaking changes

Yes

### 👀 Have you checked for similar open issues?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md)

### Are you willing to work on this issue?

Yes I am willing to submit a PR!

**Comment by prashantrai-30** [comment]
Hey @jonaslagoni,
Can I work on on this :)

**Comment by jonaslagoni** [comment]
Go for it 😄

**Comment by devilkiller-ag** [comment]
Hi @prashantrai-30, Any update on this? Feel free to ask if you got stuck somewhere. 🔥

**Comment by prashantrai-30** [comment]
Hi @devilkiller-ag ,
I started working on it but have paused for now as my exams are currently ongoing.

**Comment by devilkiller-ag** [comment]
> Hi @devilkiller-ag ,
> I started working on it but have paused for now as my exams are currently ongoing.
> 
Oh that's great! All the best for your exam🤞

**Comment by kartik-212004** [comment]
@devilkiller-ag I'm interested in working on this issue and would appreciate being assigned to it.

**Comment by NirvedMishra** [comment]
@devilkiller-ag Can I work on this or I need to be assigned for it

**Comment by prashantrai-30** [comment]
@kartik-212004 @NirvedMishra I am already working on it :)

**Comment by prashantrai-30** [comment]
Hey @devilkiller-ag ,
I am getting 404 Not Found in the Docs tab on my system, what should I do so that I can find the docs?
![Screenshot 2024-12-22 092100](https://github.com/user-attachments/assets/dacf009c-81b9-4d4f-a61c-1326ce067784)

**Comment by devilkiller-ag** [comment]
> Hey @devilkiller-ag , I am getting 404 Not Found in the Docs tab on my system, what should I do so that I can find the docs?

Hi @prashantrai-30, did you build modelina (`npm run build:modelina`) first before running `npm i` for the website? If not so, first build the modelina. Refer to the readme for the [modelina website](https://github.com/asyncapi/modelina/blob/master/modelina-website/README.md).

**Comment by prashantrai-30** [comment]
> > Hey @devilkiller-ag , I am getting 404 Not Found in the Docs tab on my system, what should I do so that I can find the docs?
> 
> Hi @prashantrai-30, did you build modelina (`npm run build:modelina`) first before running `npm i` for the website? If not so, first build the modelina. Refer to the readme for the [modelina website](https://github.com/asyncapi/modelina/blob/master/modelina-website/README.md).

still facing the same issue

**Comment by yashsaraswat2004** [comment]
Can I work on this issue??

**Comment by devilkiller-ag** [comment]
> Hey [@devilkiller-ag](https://github.com/devilkiller-ag),
> 
> I have worked on this issue and am attaching a video below for your reference. However, I am currently facing a few challenges:
> 
> 1. I am unable to see the documents in the "Docs" section.
> 2. When I click on the logo to navigate back to the home page, the theme switches back to light mode.
> 
> Could you please help me resolve these issues? Also, do you have any suggestions on how I can implement this more effectively? I would really appreciate your support.
> 
> Looking forward to your guidance.
> 
> Thank you!
> 
>  Screen.Recording.2025-03-05.150117.1.mp4

Hi @prashantrai-30, did you run `npm run build:modelina` from inside the `modelina-website` folder before running `npm run dev`?

**Comment by prashantrai-30** [comment]
Yes, I tried, but I encountered a 'Page Not Found' issue. I will try again, and I would also appreciate any suggestions you may have on how I can improve it. :)

**Comment by devilkiller-ag** [comment]
> Yes, I tried, but I encountered a 'Page Not Found' issue. I will try again, and I would also appreciate any suggestions you may have on how I can improve it. :)

Can you open a PR so that I can check it thoroughly.
